CDCgov/prime-reportstream
# Preserve MSH-18 in HL7v2->FHIR->HL7v2

This PR adds an extension to Message Header to preserve MSH-18 (character set).

Test Steps:
1. Add a non `UNICODE UTF-8` MSH-18 value on an input HL7v2 message
2. Inspect output HL7v2 and check MSH-18

## Changes
- Map MSH-18 in HL7v2 to an extension on Message Header in resulting FHIR
- Map character set extension from Message Header in FHIR to MSH-18 in resulting HL7v2

## Checklist

### Testing

## Linked Issues
- Addresses #10851
